evennia/utils/utils.py

```diff
@@ -1854,21 +1857,26 @@ def format_grid(elements, width=78, sep="  ", verbatim_elements=None):
             be inserted into the grid at the correct position and may be surrounded
             by padding unless filling the entire line. This is useful for embedding
             decorations in the grid, such as horizontal bars.
+        ignore_ansi (bool, optional): Ignore ansi markups when calculating white spacing.
 
     Returns:
         list: The grid as a list of ready-formatted rows. We return it
         like this to make it easier to insert decorations between rows, such
         as horizontal bars.
     """
+    def ansi_len(text):
+        """Use arg ignore_ansi to calculate length of text in element."""
+        return len(strip_ansi(text)) if ignore_ansi else len(text)
```

Comment:
The way to do this should really be to convert the input to `ANSIString` - you can then use `len()` on them normally without this extra function.

Reply:
The only reason I had it in a function call was to support disabling ignoring ansi characters.
Should I drop that support?

There is something I am not understanding about ANSIString.
If I change the function ansi:
```python
def ansi_len(text):
        """Use arg ignore_ansi to calculate length of text in element."""
        print(ANSIString(text).clean())
        return len(ANSIString(text).clean()) if ignore_ansi else len(text)
```

I do not receive a string with the ansi markups removed.

My output (top window is the web client) (bottom window is putty with a `evennia restart -l` running)
![image](https://user-images.githubusercontent.com/54369722/122066126-77aa5a80-cdc0-11eb-9f48-7d5c434423ed.png)